src/wp-includes/class-wp-theme.php

```diff
@@ -361,10 +361,10 @@ public function __construct( $theme_dir, $theme_root, $_child = null ) {
 		if ( ! $this->template ) {
 			$this->template = $this->stylesheet;
 			$theme_path     = $this->theme_root . '/' . $this->stylesheet;
+			$is_block_theme = $this->is_block_theme();
```

Comment:
@mukeshpanchal27 @Juzar10 This local variable is not needed. When is_block_theme is called it sets a class property. So just call this method and use this property.